src/main/java/net/snowflake/ingest/streaming/internal/StageMetadata.java

```diff
@@ -0,0 +1,127 @@
+package net.snowflake.ingest.streaming.internal;
+
+import java.util.Map;
+import net.snowflake.client.jdbc.internal.fasterxml.jackson.annotation.JsonProperty;
+
+class StageMetadata {
+  @JsonProperty("locationType")
+  private String stageType; // The stage type
+
+  @JsonProperty("location")
+  private String location; // The container or bucket
+
+  @JsonProperty("path")
+  private String path; // path of the target file
+
+  @JsonProperty("creds")
+  private Map<?, ?> credentials; // the credentials required for the  stage
+
+  @JsonProperty("region")
+  private String region; // AWS/S3/GCS region (S3/GCS only)
+
+  @JsonProperty("endPoint")
+  private String endPoint; // The Azure Storage endpoint (Azure only)
+
+  @JsonProperty("storageAccount")
+  private String storageAccount; // The Azure Storage account (Azure only)
+
+  @JsonProperty("presignedUrl")
+  private String presignedUrl; // GCS gives us back a presigned URL instead of a cred
+
+  @JsonProperty("isClientSideEncrypted")
+  private boolean isClientSideEncrypted; // whether to encrypt/decrypt files on the stage
+
+  @JsonProperty("useS3RegionalUrl")
+  private boolean useS3RegionalUrl; // whether to use s3 regional URL (AWS Only)
+
+  @JsonProperty("volumeHash")
+  private String volumeHash; // a unique id for volume assigned by server
+
+  String getStageType() {
+    return stageType;
+  }
+
+  void setLocationType(String stageType) {
+    this.stageType = stageType;
+  }
```

Comment:
getStageType but setLocationType?
A test would've caught this, which means we don't have test coverage for this part too :)

Reply:
Renamed. The serialization / deserialization was tested in `StreamingIngestStageTest.testRefreshSnowflakeMetadataRemote`. It was passed because the `@JsonProperty` override the key set by setter/getter method name, which makes the method name doesn't matter.